### PR TITLE
Add polling to builds list page

### PR DESCRIPTION
**Pull request by cadeParade**
See the most boring gif in the world:
Builds page polls for updates every 10 seconds...gif is not to time scale, but you can see it polling and updating!
poll

This PR:
- Makes a mixin to handle polling mechanics (only polls when document is in focus)
- Mixes in to components with polling to remove duplication of polling mechanics
- Converts polling part of `setup/github-app` route to a component so it can use the mixin.

### CODE REVIEW

File: app/components/build-container.js [hunk 1]
```diff
@@ -24,34 +19,13 @@ export default Component.extend({
...
-          // ember-concurrency is planning on adding a better way to cancel tasks during testing
-          // https://ember-concurrency.com/#/docs/testing-debugging
-          if (Ember.testing) {
-            this.get('runningTask').cancel();
-          }
-        });
-      yield timeout(POLLING_INTERVAL_SECONDS * 1000);
-    }
-  }).drop(),
+  pollFunction() {
```
Comment (djones) [inline app/components/build-container.js]: Perhaps `pollRefresh` instead?

File: app/components/build-container.js [hunk 2]
```diff
@@ -1,13 +1,8 @@
-import {on} from '@ember/object/evented';
 import {max, oneWay, or} from '@ember/object/computed';
 import Component from '@ember/component';
-import Ember from 'ember';
-import {task, timeout} from 'ember-concurrency';
+import PollingMixin from 'percy-web/mixins/polling';
```
Comment (djones) [inline app/components/build-container.js]: This mixin really tidies things up. Nice 👍

File: app/templates/setup/github-app.hbs [hunk 3]
```diff
@@ -1,14 +1,4 @@
...
-    <div class="Card Card--extraPadded u-text-center">
-      <h3>GitHub App processing...</h3>
-      <p>
-        {{fa-icon "fa-spinner fa-pulse"}}
-        Waiting for GitHub to complete the installation.
-      </p>
-    </div>
-  </div>
-</div>
+  {{setup-github-app installationId=installationId transitionToOrgSettings=(route-action "transitionToOrgSettings")}}
```
Comment (fotinakis) [inline app/templates/setup/github-app.hbs]: How about naming that `afterAppInstalled`? Like `afterAppInstalled=(route-action "transitionToOrgSettings")`

**Comment by djones** [comment]
It'll be interesting to see how this does with `master` now that we have the project list on the left and there are status icons that will change.

We'd want to refresh that list too on each poll update.

**Comment by cadeParade** [comment]
@djones I've checked this with the new sidebar, and the project list status bubble _does_ update when the builds are refreshed. Yay Ember.

**Comment by djones** [comment]
@cadeParade Just Works™

File: app/components/build-container.js (hunk 2 above)
Comment (fotinakis) [reply app/components/build-container.js]: Yes this is soooo 💯

**Comment by fotinakis** [comment]
Have QA'd this — it's awesome! Ship it! LGTM

**Comment by cadeParade** [comment]
Ok, I've renamed the two suggested things, will merge after the tests pass!

**Pull request merged**